### PR TITLE
Architecture detection support for windows 64/32

### PR DESCRIPTION
**Pull request by juanpicado**
I think this solve this bug https://github.com/appium/appium/issues/5289
Test included as well.

### CODE REVIEW

**Comment by jlipps** [comment]
thanks @juanpicado! I believe @soniayang97 is also fixing this bug in a different branch. we'll probably just publish that branch with the fix but i'll let her and @imurchie chime in

**Comment by juanpicado** [comment]
yw, btw i forgot give the merit to @sebv , he posted a workaround.

**Comment by imurchie** [comment]
Yes, a functionally equivalent change is [in the works](https://github.com/syang97/appium-support/blob/sonia-rewrite/lib/system.js).

If we are going to release another 1.4 patch, we could also merge this and release it in conjunction. Or we could wire in @soniayang97's  updated version when it has been done, which I think ought to be immanent (just waiting on a couple of tests).

**Comment by Jonahss** [comment]
Hm. I want to do a 1.4.8 release. Should @soniayang97's updated version just go in to that? How about Tuesday (tomorrow) for that release?

Anything else we might want in that release? @imurchie @sebv @jlipps

**Comment by Jonahss** [comment]
@moizjv wants it in.
After @soniayang97's work is merged, can you create a PR to appium to update the package.json. Then we can cherry-pick that into 1.4

File: lib/system.js [hunk 1]
```diff
@@ -15,13 +15,21 @@ exports.isLinux = function () {
   return !exports.isWindows() && !exports.isMac();
 };
 
+function isOSWin64() {
+    return process.arch === 'x64' || process.env.hasOwnProperty('PROCESSOR_ARCHITEW6432');
```
Comment (syang97) [inline lib/system.js]: should be two spaces

File: lib/system.js [hunk 2]
```diff
@@ -15,13 +15,21 @@ exports.isLinux = function () {
...
-  exec("uname -m", function (err, stdout) {
-    if (err) return cb(err);
-    if (stdout.trim() === "i686") {
-      cb(null, "32");
-    } else {
-      cb(null, "64");
-    }
-  });
+  if (exports.isWindows() ) {
+      cb(null, isOSWin64() ? "64" : "32");
```
Comment (syang97) [inline lib/system.js]: indentation

**Comment by syang97** [comment]
Besides the indentation, it pretty much looks like mine so maybe you can merge this first?

**Comment by moizjv** [comment]
@juanpicado can you fix indentation so we can totally include this for 1.4.8 :)

**Comment by juanpicado** [comment]
Sure ... In progress ..

**Comment by Jonahss** [comment]
Great!
I squashed the "fixed indentation" commit into the previous commite and merged manually :)

**Pull request closed without merging**